### PR TITLE
{vis}[foss/2021b] VTK v9.1.0 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node308.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/061d30091adfbbb9f7cac669baeeb88b for a full test report.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by branfosj** [comment]
My build failed in the `postinstallcmds` with:
```
== 2022-03-24 17:27:38,794 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd "sed "s/#VTK_VERSION#/9.1.0/" /dev/shm/build-branfosj-admin/branfosj-admin-up/VTK/9.1.0/foss-2021b/VTK-9.1.0/vtk-version.egg-info > /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/VTK/9.1.0-foss-2021b/lib/python3.9/site-packages/vtk-9.1.0.egg-info" exited with exit code 1 and output:
/bin/bash: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/VTK/9.1.0-foss-2021b/lib/python3.9/site-packages/vtk-9.1.0.egg-info: No such file or directory
```

The real problem was much earlier:
```
-- Found Python3: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/venv-dev-develop-up/bin/python3.6 (found suitable version "3.6.8", minimum required is "3.4") found components: Interpreter Development.Module Development.Embed 
```
That is that VTK is using the Python from my EB virtualenv.

File: easybuild/easyconfigs/v/VTK/VTK-9.1.0-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+]
+
+separate_build_dir = True
+
+# OpenGL
+configopts = "-DOPENGL_glu_LIBRARY=$EBROOTLIBGLU/lib/libGLU.%s " % SHLIB_EXT
+configopts += "-DOPENGL_gl_LIBRARY=$EBROOTMESA/lib/libGL.%s " % SHLIB_EXT
+configopts += "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include "
+# Python
+configopts += "-DVTK_WRAP_PYTHON=ON -DVTK_PYTHON_VERSION=3 -DVTK_PYTHON_OPTIONAL_LINK=OFF "
```
Comment (branfosj) [inline easybuild/easyconfigs/v/VTK/VTK-9.1.0-foss-2021b.eb:51]:
```suggestion
configopts += "-DVTK_WRAP_PYTHON=ON -DVTK_PYTHON_VERSION=3 -DVTK_PYTHON_OPTIONAL_LINK=OFF "
configopts += "-DPython3_ROOT_DIR=$EBROOTPYTHON "
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by branfosj** [comment]
I've set off a test build. The correct Python is detected:
```
-- Found Python3: /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Python/3.9.6-GCCcore-11.2.0/bin/python3.9 (found suitable version "3.9.6", minimum required is "3.4") found components: Interpreter Development.Module Development.Embed
```

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/6d31cb6f1033b5324feae47b57c4899d for a full test report.

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Going in, thanks @smoors!

**Pull request merged**